sdk/go/auto/stack.go

```diff
@@ -450,6 +450,102 @@ func (s *Stack) Up(ctx context.Context, opts ...optup.Option) (UpResult, error)
 	return res, nil
 }
 
+func (s *Stack) PreviewRefresh(ctx context.Context, opts ...optrefresh.Option) (PreviewResult, error) {
+	var res PreviewResult
+
+	refreshOpts := &optrefresh.Options{}
+	for _, o := range opts {
+		o.ApplyOption(refreshOpts)
+	}
+
+	args := slice.Prealloc[string](len(refreshOpts.Target))
+
+	args = debug.AddArgs(&refreshOpts.DebugLogOpts, args)
+	args = append(args, "refresh", "--preview-only")
+	if refreshOpts.Message != "" {
+		args = append(args, fmt.Sprintf("--message=%q", refreshOpts.Message))
+	}
+	if refreshOpts.ExpectNoChanges {
+		args = append(args, "--expect-no-changes")
+	}
+	for _, tURN := range refreshOpts.Target {
+		args = append(args, "--target="+tURN)
+	}
+	if refreshOpts.Parallel > 0 {
+		args = append(args, fmt.Sprintf("--parallel=%d", refreshOpts.Parallel))
+	}
+	if refreshOpts.UserAgent != "" {
+		args = append(args, "--exec-agent="+refreshOpts.UserAgent)
+	}
+	if refreshOpts.Color != "" {
+		args = append(args, "--color="+refreshOpts.Color)
+	}
```

Comment:
I wonder if we should have a helper for converting `refreshOpts` to args, that could be shared between `PreviewRefresh` and `Refresh`. The only difference being `--preview-only` vs `--yes` and `--skip-preview`.